save-core/src/commonNonJsMain/kotlin/com/saveourtool/save/core/utils/CliUtils.kt

```diff
@@ -0,0 +1,61 @@
+package com.saveourtool.save.core.utils
+
+import okio.FileSystem
+import okio.IOException
+import okio.Path.Companion.toPath
+
+import kotlinx.cli.ArgParser
+import kotlinx.cli.CLIEntity
+import kotlinx.serialization.ExperimentalSerializationApi
+import kotlinx.serialization.properties.Properties
+import kotlinx.serialization.properties.decodeFromStringMap
+
+/**
+ * Util class for processing CLI arguments
+ */
+class CliUtils {
+    companion object {
+        /**
+         * Parse properties file ([rootDir]/[projectName].properties) to [T] structure
+         *
+         * @param fs implementation of [FileSystem]
+         * @param rootDir
+         * @param projectName
+         * @return parsed [T] structure
+         * @throws IOException when [rootDir] not found
+         */
+        @OptIn(ExperimentalSerializationApi::class)
+        inline fun <reified T> parsePropertiesFile(
```

Comment:
Why not make these utils top-level methods? I actually think we had a rule in Diktat that marks classes only with methods as an anti-pattern

Reply:
It was done to be able to use it via class name: `CliUtils.parsePropertiesFile()` instead of just `parsePropertiesFile()`. For me, it increases code readable (you can see class for called method from method's call without checking imports).

But if there is a rule which prohibits public methods in object companion -- will change it to top level methods

Reply:
it's about extension for class, will refactor to extension for `FileSystem`